### PR TITLE
Add waveater module

### LINKED ISSUE #4: There should be a telegram bot
**Describe the feature or change you'd like**
The waveater telegram bot should be integrated to get and reduce the audio files from PAC(Project Audio Channnel).

The Audios should then be transcripted and stored in a Elasticsearch database.

**Additional context**
Once upon a time there was a SNEK boy called @Aichnerc. He was easily detracted by minor issues and sometimes somewhat narrow minded.
One day he was ranting about the size of our audio recordings and demanded that the size should be reduced by at least 10folde. He was whining for days until a mighty and handsome druid had mercy. The druid created a homunculus(bot), a cat girl so uwu the world has never seen before, a cat girl to serve them all, a cat girl that reduced and satisfied all the desires of programmers in the lewdest way possible. The girl was named waveater and the SNEK boys lived happily ever after. ^^

image

### PR DESCRIPTION
**Pull request by kleberbaum**
**Checklist:**

 

-   [x] Have you added an explanation of what your changes do and why you'd like them to be included?
-   [ ] Have you updated or added documentation for the change?
-   [x] Have you tested your changes with successful results?

**Type of Changes:**

 

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Documentation (non-breaking change which adds documentation)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**

- Currently there is no bot: Closes #4 

**What is the new behavior (if this is a feature change)?**

- Now there is a bot. Closes #4 

**Other information**:

- Ref: https://github.com/miraculix-org/waveater.git

### CODE REVIEW

File: esite/waveater/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# myapp/_init_.py
+default_app_config = "esite.waveater.apps.WaveaterConfig"
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2020 miraculix-org Florian Kleber
```
Comment (kleberbaum) [inline esite/waveater/__init__.py]:
```suggestion
# Copyright © 2020 miraculix-org Florian Kleber

```

File: esite/waveater/admin.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from django.contrib import admin
+
+# Register your models here.
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2020 miraculix-org Florian Kleber
```
Comment (kleberbaum) [inline esite/waveater/admin.py]:
```suggestion
# Copyright © 2020 miraculix-org Florian Kleber

```

File: esite/waveater/download_from_url.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        + ((str(milliseconds) + "ms, ") if milliseconds else "")
+    )
+    return tmp[:-2]
+
+
+# SPDX-License-Identifier: (MIT)
+# Copyright (c) 2019 Pavithran
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2020 miraculix-org Florian Kleber
```
Comment (kleberbaum) [inline esite/waveater/download_from_url.py]:
```suggestion
# Copyright © 2020 miraculix-org Florian Kleber

```

File: esite/waveater/file_handler.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+**ETA:** {time_formatter(estimated_total_time)}
+**TTC:** {time_formatter(time_to_completion) or "0s"}"""
+
+        await event.edit(progress_str)
+
+# SPDX-License-Identifier: (MIT)
+# Copyright (c) 2019 Pavithran
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2020 miraculix-org Florian Kleber
```
Comment (kleberbaum) [inline esite/waveater/file_handler.py]:
```suggestion
# Copyright © 2020 miraculix-org Florian Kleber

```

File: esite/waveater/models.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+from django.db import models
+
+# Create your models here.
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2020 miraculix-org Florian Kleber
```
Comment (kleberbaum) [inline esite/waveater/models.py]:
```suggestion
# Copyright © 2020 miraculix-org Florian Kleber

```

File: esite/waveater/tests.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+from django.test import TestCase
+
+# Create your tests here.
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2020 miraculix-org Florian Kleber
```
Comment (kleberbaum) [inline esite/waveater/tests.py]:
```suggestion
# Copyright © 2020 miraculix-org Florian Kleber

```

File: esite/waveater/views.py [hunk 7]
```diff
@@ -0,0 +1,6 @@
+from django.shortcuts import render
+
+# Create your views here.
+
+# SPDX-License-Identifier: (EUPL-1.2)
+# Copyright © 2020 miraculix-org Florian Kleber
```
Comment (kleberbaum) [inline esite/waveater/views.py]:
```suggestion
# Copyright © 2020 miraculix-org Florian Kleber

```

**Review by kleberbaum — COMMENTED** [review]
Should be ok to merge after a few changes.

File: esite/waveater/file_handler.py (hunk 4 above)
Comment (schettn) [inline esite/waveater/file_handler.py:32]: I have two questions.
1. Are two licence tags valid?
2. is miraculix-org valid?

**Review by schettn — APPROVED** [review]
lgtm

File: esite/waveater/file_handler.py (hunk 4 above)
Comment (kleberbaum) [reply esite/waveater/file_handler.py:32]: yes

**Pull request merged**